src/lib.rs

```diff
@@ -345,17 +346,25 @@ impl Build {
             // prefix, we unset `CROSS_COMPILE` for `./Configure`.
             configure.env_remove("CROSS_COMPILE");
 
-            // Infer ar/ranlib tools from cross compilers if the it looks like
-            // we're doing something like `foo-gcc` route that to `foo-ranlib`
-            // as well.
-            if path.ends_with("-gcc") && !target.contains("unknown-linux-musl") {
-                let path = &path[..path.len() - 4];
-                if env::var_os("RANLIB").is_none() {
-                    configure.env("RANLIB", format!("{}-ranlib", path));
-                }
-                if env::var_os("AR").is_none() {
-                    configure.env("AR", format!("{}-ar", path));
-                }
+            let ar = cc.get_archiver();
+            configure.env("AR", ar.get_program());
+            if ar.get_args().count() == 0 {
+                // On some platforms (like emscripten on windows), the ar to use may not be a
+                // single binary, but instead a multi-argument command like `cmd /c emar.bar`.
+                // We can't convey that through `AR` alone, and so also need to set ARFLAGS.
+                configure.env(
+                    "ARFLAGS",
+                    ar.get_args().collect::<Vec<_>>().join(OsStr::new(" ")),
+                );
+            }
+            let ranlib = cc.get_ranlib();
+            configure.env("RANLIB", ranlib.get_program());
+            if ranlib.get_args().count() == 0 {
```

Comment:
If count is 0, then we set the RANLIBFLAGS to empty string?

Why we need to do that?